### PR TITLE
feat(proxy): streaming seed discovery

### PR DESCRIPTION
**Pull request by xla**
Alleviates the need to restart the app when the seed list in the settings was updated.

### CODE REVIEW

**Review by kim — DISMISSED** [review]
lgtm

**Review by FintanH — APPROVED** [review]
:man_dancing:

**Review by cloudhead — COMMENTED** [review]
N:ice_cube:

**Review by FintanH — APPROVED** [review]
:metal:

File: proxy/api/src/main.rs [hunk 1]
```diff
@@ -86,9 +87,38 @@ async fn run(
...
+                let seeds = seed::resolve(&seeds).await.unwrap_or_else(|err| {
+                    log::error!("Error parsing seed list {:?}: {}", seeds, err);
+                    vec![]
+                });
+
+                if seeds == last_seeds {
+                    continue;
+                }
+
+                if seeds_sender.broadcast(seeds.clone()).is_err() {
```
Comment (NunoAlexandre) [inline proxy/api/src/main.rs:113]: Would be helpful to `log.error!` here, no?

**Review by NunoAlexandre — APPROVED** [review]

**Comment by rudolfs** [comment]
You might want to remove the notification then:

Screenshot 2020-10-22 at 15 59 42

https://github.com/radicle-dev/radicle-upstream/blob/67313e05a2e1ad08279e2ee70764b63d88a34cb6/ui/src/session.ts#L154

File: proxy/api/src/main.rs (hunk 1 above)
Comment (xla) [reply proxy/api/src/main.rs:113]: The error is not really helpful as it means all receivers are gone. It determines that this task should stop continuing its work. While we suffer from a complex graph of long-running routines for now, we should be most adamant about releasing resources when shutting down/reloading. Proxy's main binary is overdue for a more structured approach to plumbing taking the `Rigging` idea to the next level.

**Pull request merged**